compact_str/src/repr/boxed/mod.rs

```diff
@@ -14,6 +16,17 @@ mod inner;
 
 const MIN_SIZE: usize = core::mem::size_of::<usize>() / 2;
 
+/// [`BoxString`] grows at an amortized rates of 1.5x
+///
+/// Note: this is different than [`std::string::String`], which grows at a rate of 2x. It's debated
+/// which is better, for now we'll stick with a rate of 1.5x
+#[inline(always)]
+fn amortized_growth(cur_len: usize, additional: usize) -> usize {
+    let required = cur_len + additional;
+    let amortized = 3 * cur_len / 2;
+    core::cmp::max(amortized, required)
```

Comment:
Should this function maybe be made fallible on overflows? I.e. using checked_add?

Reply:
I'm not totally sure what we should do here yet, I left a TODO to think about the design of overflows